book/src/marshalling-types.md

```diff
@@ -0,0 +1,252 @@
+# Marshalling types
+
+An important part of embedding Gluon is translating non-primitive types from Gluon types to
+Rust types and vice versa, allowing you to seamlessly implement rich APIs with complex types.
+This translation is called _marshalling_.
+
+## Required traits
+
+Gluon provides several traits for safely marshalling types to and from Gluon code:
+
+- [VmType][] provides a mapping between Rust and Gluon types. It specifies the Gluon type
+  the implementing Rust type represents. All types that want to cross the Gluon/Rust boundary
+  must implement this trait.
+
+- [Getable][]: Types that implement `Getable` can be marshalled from Gluon to Rust. This
+  means you can use these types anywhere you are receiving values from Gluon, for example
+  as parameters for a function.
```

Comment:
Might be good to make it explicit that it is for functions implement in rust, so perhaps `for a function implemented on the Rust side`?

Reply:
Being explicit certainly won't hurt, I'll add it.